source/Halibut.Tests/UsageFixture.cs

```diff
@@ -368,22 +294,31 @@ public async Task CanSetCustomFriendlyHtmlPageHeaders()
 
         [Test]
         [System.ComponentModel.Description("Connecting over a non-secure connection should cause the socket to be closed by the server. The socket used to be held open indefinitely for any failure to establish an SslStream.")]
-        public void ConnectingOverHttpShouldFailQuickly()
+        public async Task ConnectingOverHttpShouldFailQuickly()
         {
-            var task = Task.Run(() => DoConnectingOverHttpShouldFailQuickly());
-            if (!task.Wait(5000))
-            {
-                Assert.Fail("Test did not complete within timeout");
-            }
-        }
-
-        void DoConnectingOverHttpShouldFailQuickly()
-        {
-            var services = GetDelegateServiceFactory();
-            using (var octopus = new HalibutRuntime(services, Certificates.Octopus))
+            var logger = new SerilogLoggerBuilder().Build();
+            using (var octopus = new HalibutRuntime(GetDelegateServiceFactory(), Certificates.Octopus))
             {
+                logger.Information("Halibut runtime created.");
                 var listenPort = octopus.Listen();
-                Assert.ThrowsAsync<HttpRequestException>(() => DownloadStringIgnoringCertificateValidation("http://localhost:" + listenPort));
+                logger.Information("Got port to listen on..");
+                var sw = new Stopwatch();
+                Assert.ThrowsAsync<HttpRequestException>(() =>
+                {
+                    logger.Information("Sending request.");
+                    sw.Start();
+                    try
+                    {
+                        return DownloadStringIgnoringCertificateValidation("http://localhost:" + listenPort);
+                    }
+                    finally
+                    {
+                        sw.Stop();
+                    }
+                });
```

Comment:
This test had to be improved since previously it was not measuring the thing it wanted to test.